## Задание 1.
- Скачать данные по ссылке 
- Считать данные с помощью pandas
- Вывести на экран первые 5 строк
- Посмотреть на описание признаков и на их содержание


In [1]:
import numpy as np
import pandas as pd

In [2]:
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
laptops = pd.read_csv('laptops.csv' , sep = ';')

In [4]:
laptops.head()

Company      Product   TypeName  Inches                    ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3                            1440x900   
2      HP       250 G6   Notebook    15.6                   Full HD 1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4  IPS Panel Retina Display 2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   

                          Cpu   Ram               Memory  \
0        Intel Core i5 2.3GHz   8GB            128GB SSD   
1        Intel Core i5 1.8GHz   8GB  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz   8GB            256GB SSD   
3        Intel Core i7 2.7GHz  16GB            512GB SSD   
4        Intel Core i5 3.1GHz   8GB            256GB SSD   

                            Gpu  OpSys  Weight  Price_euros  
0  Intel Iris Plus Graphics 640  macOS  1.37kg      1339.69  
1        Intel HD Graphics 6000  macOS  1.34kg       898.94  
2         Intel HD Graphics 620  No OS  1.86kg       575.00  
3            AMD Radeon Pro 455  macOS  1.83kg      2537.45  
4  Intel Iris Plus Graphics 650  macOS  1.37kg      1803.60

In [5]:
df = pd.read_csv('laptops.csv', sep=';')
df.sample(n=10)

Company                               Product            TypeName  \
278    Lenovo                     IdeaPad 320-17ISK            Notebook   
729        HP    15-BW037na (A9-9420/4GB/1TB/Radeon            Notebook   
479   Toshiba                     Portege X30-D-10J            Notebook   
1219     Acer  SP714-51 (i7-7Y75/8GB/256GB/FHD/W10)  2 in 1 Convertible   
216    Lenovo                     IdeaPad 320-17IKB            Notebook   
200      Dell                         Inspiron 7577              Gaming   
1228   Lenovo                    IdeaPad Y700-15ISK              Gaming   
204      Dell                        Precision 7520         Workstation   
685      Dell                          Alienware 15              Gaming   
266      Asus                          ZenBook Flip  2 in 1 Convertible   

      Inches                           ScreenResolution  \
278     17.3                                   1600x900   
729     15.6                          Full HD 1920x1080   
479     13.3                IPS Panel Full HD 1920x1080   
1219    14.0  IPS Panel Full HD / Touchscreen 1920x1080   
216     17.3                                   1600x900   
200     15.6            IPS Panel 4K Ultra HD 3840x2160   
1228    15.6  IPS Panel Full HD / Touchscreen 1920x1080   
204     15.6                      4K Ultra HD 3840x2160   
685     15.6                          Full HD 1920x1080   
266     13.3            Full HD / Touchscreen 1920x1080   

                              Cpu   Ram                  Memory  \
278      Intel Core i3 6006U 2GHz   4GB                 1TB HDD   
729       AMD A9-Series 9420 3GHz   4GB                 1TB HDD   
479    Intel Core i5 7200U 2.5GHz   8GB               256GB SSD   
1219    Intel Core i7 7Y75 1.3GHz   8GB               256GB SSD   
216    Intel Core i5 7200U 2.5GHz   8GB                 1TB HDD   
200   Intel Core i7 7700HQ 2.8GHz  16GB    512GB SSD +  1TB HDD   
1228  Intel Core i7 6700HQ 2.6GHz  16GB    128GB SSD +  1TB HDD   
204   Intel Xeon E3-1505M V6 3GHz  16GB    256GB SSD +  1TB HDD   
685   Intel Core i7 7820HK 2.9GHz  16GB    256GB SSD +  1TB HDD   
266    Intel Core i7 8550U 1.8GHz   8GB  512GB SSD +  512GB SSD   

                           Gpu       OpSys  Weight  Price_euros  
278      Intel HD Graphics 520  Windows 10   2.8kg       489.00  
729             AMD Radeon 520  Windows 10   2.1kg       489.00  
479      Intel HD Graphics 620  Windows 10  1.05kg      1672.00  
1219     Intel HD Graphics 615  Windows 10   1.2kg      1149.00  
216   Nvidia GeForce GTX 940MX       No OS   2.8kg       589.00  
200    Nvidia GeForce GTX 1060  Windows 10  2.65kg      1845.00  
1228   Nvidia GeForce GTX 960M  Windows 10   2.6kg      1029.00  
204        Nvidia Quadro M1200  Windows 10   2.8kg      3055.00  
685    Nvidia GeForce GTX 1070  Windows 10  3.49kg      2813.75  
266     Intel UHD Graphics 620  Windows 10   1.1kg      1499.00

In [6]:
laptops.describe()
    

Inches  Price_euros
count  1303.000000  1303.000000
mean     15.017191  1123.686992
std       1.426304   699.009043
min      10.100000   174.000000
25%      14.000000   599.000000
50%      15.600000   977.000000
75%      15.600000  1487.880000
max      18.400000  6099.000000

## Задание 2.
#### Проведите первичный анализ данных
- Изучите типы данных
- Найдите количество пропущенных ячеек в данных
- Посчитайте основные статистики по всем признакам и поизучайте

In [7]:
df.describe(include=['object'])

Company Product  TypeName   ScreenResolution  \
count     1303    1303      1303               1303   
unique      19     618         6                 40   
top       Dell  XPS 13  Notebook  Full HD 1920x1080   
freq       297      30       727                507   

                               Cpu   Ram     Memory                    Gpu  \
count                         1303  1303       1303                   1303   
unique                         118     9         39                    110   
top     Intel Core i5 7200U 2.5GHz   8GB  256GB SSD  Intel HD Graphics 620   
freq                           190   619        412                    281   

             OpSys Weight  
count         1303   1303  
unique           9    179  
top     Windows 10  2.2kg  
freq          1072    121

## Задание 3. 

#### 3.1 Ноутбуков от какой компании больше всего в наборе данных?



In [8]:
laptops['Company'].max()

'Xiaomi'

#### 3.2 Какая минимальная и максимальная стоимость у ноутбуков в данных?


In [11]:
laptops['Price_euros'].min(), laptops['Price_euros'].max()

(174.0, 6099.0)

#### 3.3 Какой самый дорогой ноутбук в данных?
Выведите все характеристики только по этому ноутбуку<br>
Если таких ноутбуков несколько, то выводите их всех


In [12]:
laptops['Company'].value_counts()

Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: Company, dtype: int64

In [17]:
laptops[laptops['Price_euros'] == laptops['Price_euros'].max()]

Company    Product TypeName  Inches                     ScreenResolution  \
196   Razer  Blade Pro   Gaming    17.3  4K Ultra HD / Touchscreen 3840x2160   

                             Cpu   Ram   Memory                      Gpu  \
196  Intel Core i7 7820HK 2.9GHz  32GB  1TB SSD  Nvidia GeForce GTX 1080   

          OpSys  Weight  Price_euros  
196  Windows 10  3.49kg       6099.0

In [18]:
laptops[laptops['Price_euros'] == laptops['Price_euros'].min()]

Company                           Product TypeName  Inches  \
1215    Acer  C740-C9QX (3205U/2GB/32GB/Chrome  Netbook    11.6   

     ScreenResolution                                   Cpu  Ram    Memory  \
1215         1366x768  Intel Celeron Dual Core 3205U 1.5GHz  2GB  32GB SSD   

                    Gpu      OpSys Weight  Price_euros  
1215  Intel HD Graphics  Chrome OS  1.3kg        174.0

In [20]:
laptops[laptops['Inches'] == laptops['Inches'].min()][['Company','TypeName','Price_euros']]


Company            TypeName  Price_euros
50    Lenovo  2 in 1 Convertible       319.00
1082  Lenovo  2 in 1 Convertible       646.27
1114  Lenovo  2 in 1 Convertible       549.00
1126  Lenovo  2 in 1 Convertible       479.00

## Задание 4.

#### 4.1 Найдите ноутбуки с самой маленькой диагональю в данных?
Выведите только следующие характеристики:
Компания
Типа ноутбука
Диагональ
Стоимость
(сделано выше)
#### 4.2 Сколько стоит самый дорогой ноутбук у компании HP?


In [21]:
laptops[laptops['Company'] == 'HP']['Price_euros'].max()

4389.0

#### 4.3 Как много ноутбуков Ultrabook с 8GB RAM?

Найдите сколько таких ультрабуков с 8GB ОЗУ в процентном соотношении относительно всех ультрабуков


In [22]:
laptops[(laptops['TypeName'] == 'Ultrabook') & (laptops['Ram'] == '8GB')].shape[0]


140

In [24]:
laptops[(laptops['TypeName'] == 'Ultrabook') & (laptops['Ram'] == '8GB')]

Company               Product   TypeName  Inches  \
0      Apple           MacBook Pro  Ultrabook    13.3   
1      Apple           Macbook Air  Ultrabook    13.3   
4      Apple           MacBook Pro  Ultrabook    13.3   
7      Apple           Macbook Air  Ultrabook    13.3   
9       Acer               Swift 3  Ultrabook    14.0   
...      ...                   ...        ...     ...   
1236      HP         EliteBook 840  Ultrabook    14.0   
1241    Asus  ZenBook UX310UA-WB71  Ultrabook    13.3   
1261    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1275    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1289    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   

                        ScreenResolution                         Cpu  Ram  \
0     IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz  8GB   
1                               1440x900        Intel Core i5 1.8GHz  8GB   
4     IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz  8GB   
7                               1440x900        Intel Core i5 1.8GHz  8GB   
9            IPS Panel Full HD 1920x1080  Intel Core i5 8250U 1.6GHz  8GB   
...                                  ...                         ...  ...   
1236                           2560x1440  Intel Core i7 6500U 2.5GHz  8GB   
1241                   Full HD 1920x1080  Intel Core i7 6500U 2.5GHz  8GB   
1261         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   
1275         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   
1289         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   

                   Memory                           Gpu       OpSys  Weight  \
0               128GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
1     128GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
4               256GB SSD  Intel Iris Plus Graphics 650       macOS  1.37kg   
7     256GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
9               256GB SSD        Intel UHD Graphics 620  Windows 10   1.6kg   
...                   ...                           ...         ...     ...   
1236            256GB SSD         Intel HD Graphics 520   Windows 7  1.54kg   
1241            256GB SSD         Intel HD Graphics 520  Windows 10  1.45kg   
1261            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1275            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1289            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   

      Price_euros  
0         1339.69  
1          898.94  
4         1803.60  
7         1158.70  
9          770.00  
...           ...  
1236      2198.19  
1241      1280.00  
1261       729.00  
1275       729.00  
1289       729.00  

[140 rows x 12 columns]

In [25]:
a = laptops[(laptops['TypeName'] == 'Ultrabook') & (laptops['Ram'] == '8GB')].shape[0]

a                                                

140

In [27]:
b = laptops[(laptops['TypeName'] == 'Ultrabook')].shape[0]
b 

196

In [29]:
100*(b-a) / b

28.571428571428573

## Задание 5.

#### 5.1 Выберите ноутбук клиенту
Клиент хочет подобрать ноутбук с 8GB или 16GB ОЗУ на Windows 10 в стоимости до 500 евро, сколько у него вариантов?


In [55]:
laptops[((laptops['Ram'] == '8GB') | (laptops['Ram'] == '16GB')) & (laptops ['OpSys'] == 'Windows 10') & (laptops['Price_euros'] < 500)].shape[0]

13

#### 5.2 Выберите ноутбук клиенту
Клиент хочет подобрать ноутбук от MSI, с видеокартой Nvidia GeForce GTX 1050 Ti и главное не с диагональю 15.6. В какой ценовой категории вышли подобные ноутбуки?


In [57]:
laptops[laptops['Cpu'] == 'Intel Core i7 7600U 2.8GHz']['Price_euros'].mean()

1915.710769230769

#### 5.3 Что дешевле?
В среднем дешевле ноутбуки с CPU Intel Core i7 7700HQ 2.8GHz или с Intel Core i7 7600U 2.8GHz?(сделано выше)


 ## Задание 6.
 
 Найдите самый легкий ноутбук
Но обратите внимание на тип и представление данных в признаке Weight, если что, замените в строке 'kg' на пустую строку через метод .str.replace()


In [59]:
laptops['Weight'] = laptops['Weight'].str.replace('kg', '')
laptops['Weight']

0       1.37
1       1.34
2       1.86
3       1.83
4       1.37
        ... 
1298     1.8
1299     1.3
1300     1.5
1301    2.19
1302     2.2
Name: Weight, Length: 1303, dtype: object